warehouse/postgres/postgresQueryBuilder.go

```diff
@@ -0,0 +1,63 @@
+package postgres
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+type QueryExecution struct {
+	txn                 *sql.Tx
+	db                  *sql.DB
+	query               string
+	enableWithQueryPlan bool
+}
+
+// handleQueryExecution
+// Print execution plan if enableWithQueryPlan is set to true else return result set.
+// Currently, these statements are supported by EXPLAIN ANALYZE
+// Any SELECT, INSERT, UPDATE, DELETE, VALUES, EXECUTE, DECLARE, CREATE TABLE AS, or CREATE MATERIALIZED VIEW AS statement, whose execution plan you wish to see.
+func handleQueryExecution(e *QueryExecution) (result sql.Result, err error) {
+	sqlStatement := e.query
+	if e.enableWithQueryPlan {
+		sqlStatement = "EXPLAIN ANALYZE " + e.query
+	}
+
+	if e.enableWithQueryPlan {
+		var rows *sql.Rows
+		if e.txn != nil {
+			rows, err = e.txn.Query(sqlStatement)
+		} else if e.db != nil {
+			rows, err = e.db.Query(sqlStatement)
+		} else {
+			err = fmt.Errorf("[WH][POSTGRES] Not able to handle query execution for statement: %s", sqlStatement)
```

Comment:
Let the error say that its because both txn and db are nil

Reply:
Done.